### DISCUSSION TITLE
How to fix "Could not read source map" error spam?

### DISCUSSION THREAD

**Discussion by Phrogz** [microsoft/vscode-discussions#470] [Extension Development QnA]
First time authoring an extension. Used `yo code` to create a Typescript extension, changed nothing in the project, and upon F5 I see 715 lines spamming the Debug Console:
```
Could not read source map for file ...
```

Per [some](https://github.com/microsoft/vscode/issues/169733) [bugs](https://github.com/microsoft/vscode/issues/131151) I've tried editing `.vscode/launch.json` to add:

```jsonc
"configurations": [
    {
        "name": "Run Extension",
        "type": "extensionHost",
        // ...
        "resolveSourceMapLocations": ["${workspaceFolder}/**", "!**/node_modules/**"]
    },
```

...but it didn't help.

I've also tried removing the `"outFiles"` section from launch.json, as [a different bug](https://github.com/microsoft/vscode/issues/106787#issuecomment-693188374) suggested it was no longer needed and would fix the problem; it does not.

Version: 1.75.1
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:35:30.018Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Linux x64 5.4.0-1091-gke
Sandboxed: Yes

**Comment by Phrogz** [comment]
Very strange...all the changes I made to `launch.json` I did by stopping debugging of the extension, changing the file, saving, and re-launching the debugger with F5. None of the changes made a difference.

To be sure, I quit and restarted VS Code...and none of the errors returned. I then tried various combinations of the above, quitting and restarting VS Code between changes...and none of the errors returned.

I've now restored the `launch.json` to the original contents, restarted VS Code...and upon starting the debugger, none of the spam appears.

It may be something like "after you build your extension for the first time, quit and relaunch VS Code to clean up this mess"?

**Edit**: _And now they're back again. This is maddening._

**Comment by r3m0t** [comment]
What files are listed in the error? I get this error about other installed extensions, but debugging of my extension works so I just ignore them.

**Reply by Phrogz** [reply]
Mostly it's all the `node_modules` extensions installed by default. But it shouldn't happen (and doesn't on macOS on 1.75.0).

**Comment by Phrogz** [comment]
The behavior does not occur with a brand new project on:

```
Version: 1.75.0
Commit: e2816fe719a4026ffa1ee0189dc89bdfdbafb164
Date: 2023-02-01T15:24:42.903Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin x64 20.6.0
Sandboxed: No
```

**Comment by olguzzar** [comment]
Since it seems like this issue has been resolved on its own, I'm going to close this discussion. Please re-open if you see this issue happen again, thanks!

**Comment by Ngocnu29** [comment]
a16885129c618c712fb3546ec171f4699750a776
